### PR TITLE
Extend ABNF to allow multi-octet values

### LINKED ISSUE #56: Current ABNF only allows for single-character baggage value
In the ABNF `value` is defined as `%x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E`, which only allows for a single octet in the value.

```
list        =  list-member 0*179( OWS "," OWS list-member )
list-member =  key OWS "=" OWS value *( OWS ";" OWS property )
property    =  key OWS "=" OWS value
property    =/ key OWS
key         =  token ; as defined in RFC 2616, Section 2.2
value       =  %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
               ; US-ASCII characters excluding CTLs,
               ; whitespace, DQUOTE, comma, semicolon,
               ; and backslash
OWS         =  *( SP / HTAB ) ; optional white space, as defined in RFC 7230, Section 3.2.3
```

The ABNF should be extended to allow for an arbitrary number of octets.

### PR DESCRIPTION
**Pull request by dyladan**
Fixes #56

***
<a href="https://pr-preview.s3.amazonaws.com/dyladan/baggage/pull/57.html" title="Last updated on Apr 20, 2021, 5:01 PM UTC (0cf91fc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/baggage/57/f39a15a...dyladan:0cf91fc.html" title="Last updated on Apr 20, 2021, 5:01 PM UTC (0cf91fc)">Diff</a>

### CODE REVIEW

**Review by SergeyKanzhelev — APPROVED** [review]
beautiful!

File: baggage/HTTP_HEADER_FORMAT.md [hunk 1]
```diff
@@ -22,16 +22,17 @@ This section uses the Augmented Backus-Naur Form (ABNF) notation of [[!RFC5234]]
...
-               ; US-ASCII characters excluding CTLs,
-               ; whitespace, DQUOTE, comma, semicolon,
-               ; and backslash
-OWS         =  *( SP / HTAB ) ; optional white space, as defined in RFC 7230, Section 3.2.3
+list           =  list-member 0*179( OWS "," OWS list-member )
+list-member    =  key OWS "=" OWS value *( OWS ";" OWS property )
+property       =  key OWS "=" OWS value
+property       =/ key OWS
+key            =  token ; as defined in RFC 2616, Section 2.2
+value          =  *baggage-octet
```
Comment (kalyanaj) [inline baggage/HTTP_HEADER_FORMAT.md:30]: Should we consider capturing the maximum byte limit for the value as part of this grammar (similar to how it is done in the "list" definition)?

Comment (dyladan) [reply baggage/HTTP_HEADER_FORMAT.md:30]: We actually can't. There is no hard limit on the `value` but on the whole list-member. You can have a long key and a short value or a short key and a long value.

Comment (kalyanaj) [reply baggage/HTTP_HEADER_FORMAT.md]: Got it, thanks.

**Review by kalyanaj — APPROVED** [review]
LGTM.

**Pull request merged**